core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/SnowflakeCallMapper.scala

```diff
@@ -428,4 +429,51 @@ class SnowflakeCallMapper extends ir.CallMapper with ir.IRHelpers {
     }
   }
 
+  private def arraySort(args: Seq[ir.Expression]): ir.Expression = {
+    makeArraySort(args.head, args.lift(1), args.lift(2))
+  }
+
+  private def makeArraySort(
+      arr: ir.Expression,
+      sortAscending: Option[ir.Expression],
+      nullsFirst: Option[ir.Expression]): ir.Expression = {
+
+    val paramSortAsc = sortAscending.getOrElse(ir.Literal.True)
+    val paramNullsFirst = nullsFirst.getOrElse {
+      paramSortAsc match {
+        case ir.Literal.True => ir.Literal.False
+        case ir.Literal.False => ir.Literal.True
+        case _ => ir.If(ir.Equals(paramSortAsc, ir.Literal.True), ir.Literal.False, ir.Literal.True)
+      }
+    }
+
+    def handleComparison(isNullOrSmallFirst: ir.Expression, nullOrSmallAtLeft: Boolean): ir.Expression = {
+      isNullOrSmallFirst match {
+        case ir.Literal.True => if (nullOrSmallAtLeft) ir.Literal(-1) else oneLiteral
+        case ir.Literal.False => if (nullOrSmallAtLeft) oneLiteral else ir.Literal(-1)
+        case _ =>
+          if (nullOrSmallAtLeft) ir.If(ir.Equals(isNullOrSmallFirst, ir.Literal.True), ir.Literal(-1), oneLiteral)
+          else ir.If(ir.Equals(isNullOrSmallFirst, ir.Literal.True), oneLiteral, ir.Literal(-1))
+      }
+    }
+
+    val comparator = ir.LambdaFunction(
+      ir.Case(
+        None,
+        Seq(
+          ir.WhenBranch(ir.And(ir.IsNull(ir.Id("left")), ir.IsNull(ir.Id("right"))), zeroLiteral),
+          ir.WhenBranch(ir.IsNull(ir.Id("left")), handleComparison(paramNullsFirst, nullOrSmallAtLeft = true)),
+          ir.WhenBranch(ir.IsNull(ir.Id("right")), handleComparison(paramNullsFirst, nullOrSmallAtLeft = false)),
+          ir.WhenBranch(
+            ir.LessThan(ir.Id("left"), ir.Id("right")),
+            handleComparison(paramSortAsc, nullOrSmallAtLeft = true)),
+          ir.WhenBranch(
+            ir.GreaterThan(ir.Id("left"), ir.Id("right")),
+            handleComparison(paramSortAsc, nullOrSmallAtLeft = false))),
+        Some(zeroLiteral)),
+      Seq(ir.UnresolvedNamedLambdaVariable(Seq("left")), ir.UnresolvedNamedLambdaVariable(Seq("right"))))
+
+    ir.ArraySort(arr, comparator)
```

Comment:
I would move this as optimizer rule, which applies the rules

https://docs.databricks.com/en/sql/language-manual/functions/array_sort.html